File: src/types/async/async.ts
const delayCallback = (callback: () => void, wait: number) => async () => {
  await new Promise((resolve) => setTimeout(resolve, wait));
  callback();
};

const printOneDelay = delayCallback(() => {
  console.log("One");
}, 1000);

printOneDelay();
console.log("Two");

File: src/types/async/async.go
package main

import (
	"fmt"
	"time"
)

func delayCallback(callback func(), c chan struct{}, wait float64) {
	time.Sleep(time.Millisecond * time.Duration(wait))
	callback()
	c <- struct{}{}
}

func main() {
	c := make(chan struct{})

	go delayCallback(func() { fmt.Println("One") }, c, 1000)
	fmt.Println("Two")

	<-c
}
